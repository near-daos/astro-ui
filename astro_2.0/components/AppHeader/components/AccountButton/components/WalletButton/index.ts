export * from 'astro_2.0/components/AppHeader/components/AccountButton/components/WalletButton/WalletButton';
