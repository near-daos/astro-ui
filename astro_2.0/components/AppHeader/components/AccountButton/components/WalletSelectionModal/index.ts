export * from 'astro_2.0/components/AppHeader/components/AccountButton/components/WalletSelectionModal/WalletSelectionModal';
