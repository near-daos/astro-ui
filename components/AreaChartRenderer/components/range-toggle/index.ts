export * as RangeToggle from './RangeToggle';
