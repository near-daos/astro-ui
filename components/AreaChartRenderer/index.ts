import {
  AreaChartRenderer,
  AreaChartProps,
} from 'components/AreaChartRenderer/AreaChartRenderer';

export default AreaChartRenderer;
export type { AreaChartProps };
