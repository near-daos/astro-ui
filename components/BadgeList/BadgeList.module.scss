.selected {
  display: flex;
  flex: 1;
}

.selected .selectedWrapper {
  margin-right: 4px;
}

.collapsedLabel {
  visibility: hidden;
}

.collapsedLabel.visible {
  visibility: visible;
}
