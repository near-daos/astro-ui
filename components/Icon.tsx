import React from 'react';
import getConfig from 'next/config';

import * as icons from 'assets/icons';

export type IconName = keyof typeof icons;

export type IconProps = React.SVGProps<SVGSVGElement> & {
  name: IconName;
  title?: string;
};

export const Icon: React.VFC<IconProps> = ({ name, title, ...svgProps }) => {
  const { viewBox, id, url } = icons[name];

  const config = getConfig();

  return (
    <svg viewBox={viewBox} {...svgProps}>
      {title && <title> {title}</title>}
      <use
        href={
          config?.publicRuntimeConfig?.shouldUseExternalAssetUrl
            ? `${config?.publicRuntimeConfig?.assetUrl}#${id}`
            : `/_next/${url}`
        }
      />
    </svg>
  );
};
