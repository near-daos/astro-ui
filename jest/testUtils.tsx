import {
  RenderResult,
  render as reactTestingLibRender,
} from '@testing-library/react';
import { ReactNode } from 'react';
import { NextRouter } from 'next/router';
import { RouterContext } from 'next/dist/shared/lib/router-context';

function createMockRouter(router?: Partial<NextRouter>): NextRouter {
  return {
    basePath: '',
    pathname: '/',
    route: '/',
    query: {},
    asPath: '/',
    back: jest.fn(),
    beforePopState: jest.fn(),
    prefetch: jest.fn(),
    push: jest.fn(),
    reload: jest.fn(),
    replace: jest.fn(),
    events: {
      on: jest.fn(),
      off: jest.fn(),
      emit: jest.fn(),
    },
    isFallback: false,
    isLocaleDomain: false,
    isReady: true,
    defaultLocale: 'en',
    domainLocales: [],
    isPreview: false,
    ...router,
  };
}

export function render(
  component: ReactNode,
  router?: Partial<NextRouter>
): RenderResult {
  return reactTestingLibRender(
    <RouterContext.Provider value={createMockRouter(router)}>
      {component}
    </RouterContext.Provider>
  );
}
