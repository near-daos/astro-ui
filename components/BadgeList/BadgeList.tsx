import React, { FC } from 'react';
import cn from 'classnames';
import { Badge } from 'components/Badge';
import styles from './BadgeList.module.scss';

type Item = {
  label: string;
  component: JSX.Element;
};

interface BadgeListProps {
  selectedItems: Item[];
  showPlaceholder: boolean;
}

export const BadgeList: FC<BadgeListProps> = ({
  selectedItems,
  showPlaceholder,
}) => {
  return (
    <div className={styles.selected}>
      {selectedItems
        .filter((k, i) => {
          if (i === 0) {
            return true;
          }

          return !showPlaceholder;
        })
        .map(sel => {
          return (
            <div key={sel.label} className={styles.selectedWrapper}>
              {sel.component}
            </div>
          );
        })}
      <div
        className={cn(styles.collapsedLabel, styles.selectedWrapper, {
          [styles.visible]: showPlaceholder,
        })}
      >
        <Badge size="small" variant="turqoise">
          +{selectedItems.length - 1}
        </Badge>
      </div>
    </div>
  );
};
