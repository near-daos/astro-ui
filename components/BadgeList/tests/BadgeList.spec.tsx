import { render } from 'jest/testUtils';

import { BadgeList } from 'components/BadgeList';

describe('Badge List', () => {
  it('Should render component', () => {
    const items = [
      {
        label: 'Hello',
        component: <div>Hello</div>,
      },
      {
        label: 'World',
        component: <div>World</div>,
      },
    ];

    const { container } = render(
      <BadgeList selectedItems={items} showPlaceholder />
    );

    expect(container).toMatchSnapshot();
  });
});
