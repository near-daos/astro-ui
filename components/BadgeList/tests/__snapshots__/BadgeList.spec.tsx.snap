// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge List Should render component 1`] = `
<div>
  <div
    class="selected"
  >
    <div
      class="selectedWrapper"
    >
      <div>
        Hello
      </div>
    </div>
    <div
      class="collapsedLabel selectedWrapper visible"
    >
      <div
        class="badge small turqoise"
      >
        +
        1
      </div>
    </div>
  </div>
</div>
`;
